packages/cojson/src/coValues/coMap.ts

```diff
@@ -39,44 +40,107 @@ export class RawCoMapView<
   /** @category 6. Meta */
   core: CoValueCore;
   /** @internal */
-  ops: {
+  latest: {
+    [Key in keyof Shape & string]?: MapOp<Key, Shape[Key]>;
+  };
+  /** @internal */
+  latestTxMadeAt: number;
+  /** @internal */
+  cachedOps?: {
     [Key in keyof Shape & string]?: MapOp<Key, Shape[Key]>[];
   };
   /** @internal */
+  options?: { ignorePrivateTransactions: boolean; atTime?: number };
+  /** @internal */
   atTimeFilter?: number = undefined;
   /** @category 6. Meta */
   readonly _shape!: Shape;
 
   /** @internal */
   constructor(
     core: CoValueCore,
-    options?: { ignorePrivateTransactions: true },
+    options?: { ignorePrivateTransactions: boolean; atTime?: number },
   ) {
     this.id = core.id as CoID<this>;
     this.core = core;
-    this.ops = {};
+    this.latest = {};
+    const latest = this.latest;
+    this.latestTxMadeAt = 0;
+    this.options = options;
+
+    const validTransactions = core.getValidSortedTransactions(options);
+
+    for (let i = validTransactions.length - 1; i >= 0; i--) {
```

Comment:
By default, we only collect the latest change for each key now. We iterate backwards over transactions because then we only need to set the entry for each key once (since they are sorted by madeAt already)